strapi/strapi
# fix: drop and create columns and indexes in the correct order

### What does it do?

fixes #21398 

### Why is it needed?

When we were dropping a column in the database, we were still trying to delete indexes on that column, but those indexes were already being dropped along with the column, so we got an error

NOTE: specifically for indexes, I think we should consider in the future adding a conditional check for if the index exists, and if so log a warning rather than throwing an error. Bugs like this one would still be caught and fixed, but it would become a 'low' rather than a 'critical' or 'high' because Strapi would still be able to start up, and the worst case scenario would be a few extra indexes lingering when they shouldn't be there

### How to test it?

With all supported databases:

- Specifically, test the linked issue
- More generally, test all kinds of creating and deleting content types and relations in the CTB
- To be 100% safe, please test a migration of a complex dataset (ie, one with every relation type) from v4 to v5 and ensure this hasn't broken anything there

I will also add a CTB e2e test for this asap before we merge, but it may not be able to be enabled on the CI due to issues we haven't yet been able to solve with github and the server restarts required by the CTB

### Related issue(s)/PR(s)

Let us know if this is related to any issue/pull request
